domain/modelconfig/validators/validators.go

```diff
@@ -180,30 +180,9 @@ type SecretBackendProvider interface {
 // is performed. Any validation errors will satisfy config.ValidationError.
 func SecretBackendChecker(provider SecretBackendProvider) config.ValidatorFunc {
 	return func(ctx context.Context, cfg, old *config.Config) (*config.Config, error) {
-		backendName := cfg.SecretBackend()
-		if backendName == old.SecretBackend() {
+		if cfg.SecretBackend() == "" {
 			return cfg, nil
 		}
-		if backendName == "" {
-			return cfg, &config.ValidationError{
-				InvalidAttrs: []string{config.SecretBackendKey},
-				Reason:       "secret back cannot be empty",
-			}
-		}
-		if backendName == config.DefaultSecretBackend {
-			return cfg, nil
-		}
-
-		has, err := provider.HasSecretsBackend(backendName)
-		if err != nil {
-			return cfg, fmt.Errorf("fetching secret backend for %q to validate model config: %w", backendName, err)
-		}
-		if !has {
-			return cfg, &config.ValidationError{
-				InvalidAttrs: []string{config.SecretBackendKey},
-				Reason:       fmt.Sprintf("secret backend %q not found", backendName),
-			}
-		}
-		return cfg, nil
+		return nil, fmt.Errorf("cannot set model secret backend via model config anymore")
```

Comment:
This should be a typed `NotSupported` error.
Need to remember to update 3.6 CLI - the CLI should catch the typed error and print a suitable message telling the user to use `model-secret-backend` instead (backport model-secret-backend cli to 3.6).
This is to allow a 3.6 cli to work with 3.x or 4.x controllers.

This branch also needs to remove secret backend from model config schema